python/tank/bootstrap/manager.py

```diff
@@ -60,22 +61,110 @@ def __repr__(self):
         """
         The pipeline configuration that is being operated on.
         By default, the primary pipeline config will be used.
         """
-        self._pipeline_configuration_name = name
+        return self._pipeline_configuration_name
 
-    def _get_pipeline_configuration(self):
+    def _set_pipeline_configuration(self, name):
         """
         The pipeline configuration that is being operated on.
         By default, the primary pipeline config will be used.
         """
-        return self._pipeline_configuration_name
+        self._pipeline_configuration_name = name
 
     pipeline_configuration = property(_get_pipeline_configuration, _set_pipeline_configuration)
 
+
+    def _get_do_shotgun_config_lookup(self):
+        """
+        Flag to indicate if the bootstrap process should connect to
+        Shotgun and attempt to resolve a config. Defaults to True.
+
+        If ``True``, the bootstrap process will connect to Shotgun as part
+        of the startup, look for a pipeline configuration and attempt
+        to resolve a toolkit environment to bootstrap into via the
+        Pipeline configuration data. Failing this, it will fall back on
+        the :meth:`base_configuration`.
+
+        If ``False``, no Shotgun lookup will happen. Instead, whatever config
+        is defined via :meth:`base_configuration` will always be used.
+        """
+        return self._do_shotgun_config_lookup
+
+    def _set_do_shotgun_config_lookup(self, status):
+        """
+        Flag to indicate if the bootstrap process should connect to
+        Shotgun and attempt to resolve a config. Defaults to True.
+
+        If True, the bootstrap process will connect to Shotgun as part
+        of the startup, look for a pipeline configuration and attempt
+        to resolve a toolkit environment to bootstrap into via the
+        Pipeline configuration data. Failing this, it will fall back on
+        the :meth:`base_configuration`.
+
+        If False, no Shotgun lookup will happen. Instead, whatever config
+        is defined via :meth:`base_configuration` will always be used.
+        """
+        self._do_shotgun_config_lookup = status
+
+    do_shotgun_config_lookup = property(_get_do_shotgun_config_lookup, _set_do_shotgun_config_lookup)
+
+
+    def _get_entry_point(self):
+        """
+        The entry point defines the scope of the bootstrap operation.
+
+        If you are bootstrapping into an entire Toolkit pipeline, e.g
+        a traditional Toolkit setup, this should be left blank.
+
+        If you are writing a plugin that is intended to run side by
+        side with other plugins in your target environment, the entry
+        point will be used to define a scope and sandbox in which your
+        plugin will execute. For example, if your plugin is packaging up
+        review tools for RV, name your entry point "RV Review". If your plugin
+        embeds the shotgun panel in a Maya Plugin, name it "Maya Panel".
```

Comment:
It should be clear that this is intended *future* behavior and that this isn't actually what you'll currently get.
:)